### PR TITLE
add support for new Order and OrderItem entities

### PR DESCRIPTION
**Pull request by AntonKurylo**

### CODE REVIEW

File: src/main/java/mate/academy/controller/CategoryController.java [hunk 1]
```diff
@@ -43,8 +43,8 @@ public CategoryDto createCategory(@RequestBody @Valid CreateCategoryRequestDto r
             description = "Returns a list of all available categories")
     @PreAuthorize("hasRole('ROLE_USER')")
     @GetMapping
-    public List<CategoryDto> findAllCategories() {
-        return categoryService.findAll();
+    public List<CategoryDto> findAllCategories(Pageable pageable) {
```
Comment (sarakhmen) [inline src/main/java/mate/academy/controller/CategoryController.java:46]: please add default parameters for your pageable

File: src/main/java/mate/academy/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    @PostMapping
+    public OrderDto createOrder(@RequestBody @Valid CreateOrderRequestDto requestDto) {
+        return orderService.save(requestDto);
+    }
+
+    @Operation(summary = "Get all orders by user",
+            description = "Returns a list of all user orders")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping
+    public List<OrderDto> findAllOrdersByUser(Pageable pageable) {
```
Comment (sarakhmen) [inline src/main/java/mate/academy/controller/OrderController.java:42]: same here

Comment (sarakhmen) [reply src/main/java/mate/academy/controller/OrderController.java:42]: please fix in all places

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @Override
+    public List<OrderDto> findAllByUser(Pageable pageable) {
+        return orderRepository.findAllByUserId(authService.getAuthenticatedUserId(), pageable)
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
```
Comment (sarakhmen) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java:56]: add transactional

**Review by sarakhmen — CHANGES_REQUESTED** [review]
let's improve your solution :)

File: src/main/java/mate/academy/controller/CategoryController.java (hunk 1 above)
Comment (AntonKurylo) [reply src/main/java/mate/academy/controller/CategoryController.java:46]: For pagination, the _Pageable_ object contains the attributes: _page_ and _size_. If you do not pass these fields, the default values are _page = 0_ and _size = 20_.

Or should I use the _@PageableDefault_ annotation?

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java (hunk 3 above)
Comment (AntonKurylo) [reply src/main/java/mate/academy/service/impl/OrderServiceImpl.java:56]: The _@.Transactional_ combines multiple database writes into a single atomic transaction.
In my method, there is only one write to the database, so why would I use _@.Transactional_?

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    public UpdateOrderResponseDto updateById(Long id, UpdateOrderRequestDto requestDto) {
+        Order updatedOrder = orderRepository.findById(id)
+                .map(order -> {
+                    order.setStatus(requestDto.status());
+                    return order;
+                }).orElseThrow(() ->
+                        new EntityNotFoundException("Cannot find an order by id: " + id));
```
Comment (okuzan) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: this looks messy TBH
```suggestion
    Order updatedOrder = orderRepository.findById(id)
            .orElseThrow(() -> new EntityNotFoundException("Cannot find an order by id: " + id));
            
        updatedOrder.setStatus(requestDto.status());
        Order savedOrder = orderRepository.save(updatedOrder);
        return orderMapper.toUpdateDto(savedOrder);
```
I would write it like this

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java (hunk 3 above)
Comment (okuzan) [reply src/main/java/mate/academy/service/impl/OrderServiceImpl.java:56]: agree

File: src/main/java/mate/academy/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        return orderMapper.toUpdateDto(orderRepository.save(updatedOrder));
+    }
+
+    @Override
+    public List<OrderItemDto> findAllOrderItemsByOrderId(Long orderId, Pageable pageable) {
+        Long authUserId = authService.getAuthenticatedUserId();
+        return orderItemRepository
+                .findAllByOrderIdAndUserId(orderId, authUserId, pageable).stream()
+                .map(orderMapper::toOrderItemDto)
```
Comment (okuzan) [inline src/main/java/mate/academy/service/impl/OrderServiceImpl.java]: FYI you can define mapping of list to list in mapstruct mapper with one line

**Review by okuzan — APPROVED** [review]

**Pull request merged**